book/animations.md

```diff
@@ -370,235 +421,287 @@ browsers.
 
 :::
 
-Opacity animations
-==================
-
-In Chapter 12 we [implemented](scheduling.md#animating-frames) the
-`requestAnimationFrame` API and built a demo that modifies the `innerHTML`
-of a DOM element on each frame. That is indeed an animation---a
-*JavaScript-driven* animation---but `innerHTML` is generally not how
-high-quality animations are done on the web. Instead, it almost always makes
-sense to first render some content into layout objects, then apply a DOM
-animation to the layout tree by animating CSS properties.
-
-It's straightforward to see how this works for opacity: define some
-HTML that you want to animate, then interpolate the `opacity` CSS property in
-JavaScript smoothly from one value to another. Here's an example that animates
-from 0.999[^why-not-one] to 0.1, over 120 frames (about two seconds), then back
-up to 0.999 for 120 more frames, and repeats.
-
-[^why-not-one]: The animation starts below 1 because most real browsers remove
-certain animation-related optimizations when opacity is exactly 1. So it's
-easier to dig into the performance of this example on a real browser with 0.999
-opacity. Starting animations at 0.999 is also a common trick used on web sites
-that want to avoid visual popping of the content as it goes in and out of
-GPU-accelerated mode. I chose 0.999 because the visual difference from 1.0 is
-imperceptible.
-
-The HTML is:
+Compositing
+===========
 
-``` {.html file=example-opacity-html}
-<div>Test</div>
-```
+So, how do we do less work in the raster and draw phase? The answer is
+a technique called *compositing*, which just means caching some
+rastered images on the GPU and reusing them during later
+frames.[^compositing-def]
 
-And here is the `animate` implementation for this example:
-``` {.javascript file=example-opacity-js}
-var frames_remaining = 120;
-var go_down = true;
-var div = document.querySelectorAll("div")[0];
-function animate() {
-    var percent_remaining = frames_remaining / 120;
-    if (!go_down) percent_remaining = 1 - percent_remaining;
-    div.style = "opacity:" +
-        (percent_remaining * 0.999 +
-            (1 - percent_remaining) * 0.1);
-    if (frames_remaining-- == 0) {
-        go_down = !go_down
-        frames_remaining = 120;
-    }
-    return true;
-}
+[^compositing-def]: The term [*compositing*][compositing] just means
+combining multiple images together into a final output. In browsers,
+we're typically combining rastered images into the final image of the
+page, but a similar technique is used in many operating systems to
+combine the contents of multiple windows. "Compositing" can also refer
+to multi-threaded rendering.
+
+[compositing]: https://en.wikipedia.org/wiki/Compositing
+
+To explain compositing, we'll need to think about our browser's
+display tree. Printing it is going to require similar code in every
```

Comment:
```suggestion
display list. Printing it is going to require similar code in every
```

Reply:
It is a tree (a forest actually), but we call it a display list anyway throughout.